packages/core/src/app_config.ts

```diff
@@ -134,8 +134,8 @@ const dumpConfig = async (config: AppConfig): Promise<void> => (
 
 const mergeConfigWithEnv = async (config: AppConfig): Promise<AppConfig> => {
   config.telemetry = {
-    token: telemetryToken(),
-    url: telemetryURL(),
+    token: config.telemetry.token || telemetryToken(),
```

Comment:
when checking for `undefined` use `??`, not `||`
in this case, seems like this should be
```suggestion
    token: config.telemetry.token ?? telemetryToken(),
```